### ISSUE TITLE
Lighthouse should be a peer dependency + Support for LH 11

### ISSUE THREAD

**Issue by ittaibaratz** [abhinaba-ghosh/playwright-lighthouse#69]
Hello @abhinaba-ghosh and @ankur-lt 

I am seeing a regression due to https://github.com/abhinaba-ghosh/playwright-lighthouse/issues/63 and https://github.com/abhinaba-ghosh/playwright-lighthouse/pull/64.

With Lighthouse 10 being a dependency, since I am using Lighthouse 11, I now get two copies of LH installed. One under node_modules (11) and one under playwright-lighthouse/node_modules/lighthouse (10). This leads to some weird LH errors, i.e.:
```
Error: Failed to find dependency "DevtoolsLog" for "IFrameResource" artifact
Check that...
1. A gatherer exposes a matching Symbol that satisfies "DevtoolsLog".
2. "DevtoolsLog" is configured to run before "IFrameResource"
    at throwInvalidDependencyOrder (file:///home/jenkins/workspace/my-project/node_modules/playwright-lighthouse/node_modules/lighthouse/core/config/validation.js:287:9)
```

I have 2 suggestions for resolving this:

1. LH should be both a devDependency and a peerDependency.
2. The peerDependency should allow both versions 10 and 11, i.e.
```
  "dependencies": {
    "chalk": "^4.1.2",
    "ua-parser-js": "^1.0.2"
  },
  "devDependencies": {
    "@types/mocha": "9.1.0",
    "@types/node": "^16.11.26",
    "chai": "^4.3.6",
    "eslint": "^8.12.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-prettier": "^4.0.0",
    "lighthouse": "^10.0.1",
    "mocha": "9.2.2",
    "prettier": "^2.6.1"
  },
  "peerDependencies": {
    "lighthouse": ">= 10.0.1",
    "playwright-core": "^1.19.2"
  },
```

Also, since this project works well with LH 11 (At least per my experience), any reason to not just update to LH 11? 

I am happy to do a PR myself, just let me know if this solution is acceptable.

**Comment by abhinaba-ghosh** [comment]
I dont see a reason not doing this. Kindly raise a PR. I think we need to update this package version as per the major version update from loghthouse so that anyone can using this package version as per their lighthouse version without breaking their workflows.

**Comment by ittaibaratz** [comment]
> I dont see a reason not doing this. Kindly raise a PR. I think we need to update this package version as per the major version update from loghthouse so that anyone can using this package version as per their lighthouse version without breaking their workflows.

Thanks!

I did https://github.com/abhinaba-ghosh/playwright-lighthouse/pull/70 - It also required a Node upgrade and I did a few other things on the way. I think the package should also use `@playwright/test` instead of `playwright-core`, I can do it in a follow up PR. Please review and provide feedback. Have a great weekend!

**Comment by abhinaba-ghosh** [comment]
the changes are available in `v4.0.0`
